api/lib/models/location/location-data-mapper-impl.ts

```diff
@@ -0,0 +1,170 @@
+      .toParam();
+    query.text = query.text.concat(';');
+    return from(
+      this.sql.query(query.text, query.values, { stream: false, cache: false }),
+    )
+      .pipe(map(() => ({ result: 'Success', data: undefined })))
+      .toPromise();
+  }
+
+  public get(id: UidType, opts?: IUowOpts): Promise<IDbResult<Location>> {
+    let queryBuilder = squel
+      .select({ autoQuoteFieldNames: true, autoQuoteTableNames: true })
+      .from(this.tableName);
+    if (opts && opts.fields) {
+      queryBuilder = queryBuilder.fields(opts.fields);
+    }
+    queryBuilder = queryBuilder.where(`${this.pkColumnName}= ?`, id);
+    const query = queryBuilder.toParam();
+    query.text = query.text.concat(';');
+    return from(
+      this.sql.query<Location>(query.text, query.values, {
+        cache: true,
+        stream: false,
+      }),
+    )
+      .pipe(
+        map((location: Location) => ({ result: 'Success', data: location })),
+      )
+      .toPromise();
+  }
+
+  public getAll(): Promise<IDbResult<Stream<Location>>> {
+    const query = squel
+      .select({ autoQuoteTableNames: true, autoQuoteFieldNames: true })
+      .from(this.tableName, 'location')
+      .field('location.*')
+      .field('location.location_name')
+      .join('HACKATHON', 'h', 'h.uid=location.hackathon and h.active=true')
+      .toString()
+      .concat(';');
+    return from(
+      this.sql.query<Location>(query, [], { stream: true, cache: true }),
+    )
+      .pipe(
+        map((location: Stream<Location>) => ({
+          data: location,
+          result: 'Success',
+        })),
+      )
+      .toPromise();
+  }
+
+  public getCount(): Promise<IDbResult<number>> {
+    const query = squel
+      .select({ autoQuoteTableNames: true, autoQuoteFieldNames: false })
+      .from(this.tableName)
+      .field(`COUNT(${this.pkColumnName})`, 'count')
+      .toString()
+      .concat(';');
+    const params = [];
+    return from(
+      this.sql.query<number>(query, params, { stream: true, cache: true }),
+    )
+      .pipe(map((result: number) => ({ result: 'Success', data: result })))
+      .toPromise();
+  }
+
+  public insert(object: Location): Promise<IDbResult<Location>> {
+    const validation = object.validate();
+    if (!validation.result) {
+      this.logger.warn('Validation failed while adding object.');
+      this.logger.warn(object.dbRepresentation);
+      return Promise.reject({
+        data: new HttpError(validation.error, 400),
+        result: 'error',
+      });
+    }
+    const query = squel
+      .insert({ autoQuoteFieldNames: true, autoQuoteTableNames: true })
+      .into(this.tableName)
+      .setFieldsRows([object.dbRepresentation])
+      .toParam();
+    query.text = query.text.concat(';');
+    return from(
+      this.sql.query<void>(query.text, query.values, {
+        cache: false,
+        stream: false,
+      }),
+    )
+      .pipe(map(() => ({ result: 'Success', data: object })))

```

Comment:
instead of `data: object`, use `data:  object.cleanRepresentation`. We recently found that by itself the object contains certain properties we would like to not send over the network